### DISCUSSION TITLE
For some reason the width and height are swapped. Cannot re-swapp the image.

### DISCUSSION THREAD

**Discussion by shanji97** [dlemstra/Magick.NET#1489] [Help]
So for some reason the MagickImage object width and height are swapped. As I try to manually swap it nothin seems to work, even though i have the Exif orientation data to determine to dimensions. 
```
private static bool IsSwappNeeded(IExifProfile profile)
  {
    foreach (var exifValue in profile.Values)
    {
      if (exifValue.Tag.Equals(ExifTag.Orientation))
      {
        var orientationValue = exifValue.GetValue().ToString();
        var orientation = byte.Parse(orientationValue);
        return orientation == (byte)ImageOrientationPropertyEnum.Rotate90CW ||
          orientation == (byte)ImageOrientationPropertyEnum.Rotate270CW;
      }
    }
    return false;
  }
```
I also would like to resize the image, but I get an image which is a cut off on the height coordinate.
```
using (MemoryStream ms = new MemoryStream(attachmentData))
{
  MagickImage magickImage = new MagickImage(ms);
/*Other processing*/
  (var actualWidth, var actualHeight) = IsSwappNeeded(magickImage.GetExifProfile())? (magickImage.Height,magickImage.Width):(magickImage.Width,magickImage.Height);
  (var destWidth,var destHeight) = (actualWidth, actualHeight);

  if (destWidth > MaxImageWidth && destHeight != 0) //MaxImageWidth is 2048
  {
    destWidth = MaxImageWidth;
    //var ratio = (float)magickImage.Width / magickImage.Height;
    var ratio = (float)actualWidth / actualHeight;
    destHeight = (int)(MaxImageWidth / ratio);
  }
  magickImage.FilterType = FilterType.Lanczos;
  magickImage.Resize(destWidth, destHeight);

  return magickImage;
}
```
I also tried the rotation code, but it is still cut off. This issue only seems to occur when using images take from Samsung devices in portrait mode and Canon devices, for which I do not have any idea how they are orientated.  Any idea how to accordingly handle those images (with Rotate 90 CW and Rotate 270 CW tags included)?
Also I forgot to mention, that I am using .net framework 4.8 with Magick .net core version 13.1.3 and Magick .net q8 AnyCpu version 13.1.3

**Comment by dlemstra** [comment]
Have you tried to using `magickImage.AutoOrient()` ?

**Reply by shanji97** [reply]
No, currently I played with  `.Resize()`, .Rotate() and `.Scale()` methods, but I'll try with this method too.
![Missing image part](https://github.com/dlemstra/Magick.NET/assets/13587507/6c90303f-2736-4ff0-9e4f-e57c2b417f3c)
![20231222_073130](https://github.com/dlemstra/Magick.NET/assets/13587507/f5899d31-b316-4983-a9fd-a0814fc4009c)
The current snippet to check the exif tag is the following.
```
private static bool IsSwapNeeded(IExifProfile profile, out byte detectedRotation)
{
  detectedRotation = 0;
  if (profile is null)
    return false;

  foreach (var exifValue in profile.Values)
  {
    if (exifValue.Tag.Equals(ExifTag.Orientation))
    {
      var orientationValue = exifValue.GetValue().ToString();
      var orientation = byte.Parse(orientationValue);
      detectedRotation = orientation;
      return orientation.Equals((byte)ImageOrientationPropertyEnum.Rotated90CW) ||
        orientation.Equals((byte)ImageOrientationPropertyEnum.Rotated270CW);
    }
  }
  return false;
}
```
The problematic part is when I get 6 or 8 in the "detectedOrientation" variable which corresponds to the Clowise rotation of 90 or 270 degrees, which is embedded on Samsung portrait images, some Cannon images and probably somewhere else. If I use `.Rotate(270)` I get the following image, which is not cropped, but rotated and one button click away I can have an almost correct result, but still this should not be the workaround here. 
![image](https://github.com/dlemstra/Magick.NET/assets/13587507/862f4c8b-9936-46ef-8ebe-9755aac9545c)
Unfortunately I cannot upgrade since doing so will throw me an exception.
![image](https://github.com/dlemstra/Magick.NET/assets/13587507/fe20c222-5e90-4dee-822b-6d5d5ff2a35e)
The code snippet I currently use is, but most different combinations produce the same result.
```
using (MemoryStream ms = new MemoryStream(attachmentData))
{
  MagickImage magickImage = new MagickImage(ms);

  magickImage.AnimationDelay = 0;
  magickImage.AnimationIterations = 1;
  magickImage.Page = new MagickGeometry(page);

  // Swap height and width if the Rotate 90 CW or Rotate 270 CW Exif tag is present. Keep in mind, that the coordinates are swapped in .net interpretation for some reason.
  IsSwapNeeded(magickImage.GetExifProfile(), out byte detectedRotation);

  //(var actualWidth, var actualHeight) = IsSwapNeeded(magickImage.GetExifProfile(), out byte detectedRotation) ? (magickImage.Height, magickImage.Width) : (magickImage.Width, magickImage.Height);
  //(var destWidth,var destHeight) = (actualWidth, actualHeight);
  int destHeight = magickImage.Height;
  int destWidth = magickImage.Width;

  if (detectedRotation != 0)
    rotation = detectedRotation;
  switch (rotation)
  {
    case (byte)ImageOrientationPropertyEnum.Rotated270CW:
      magickImage.Rotate(0);
      break;
    case (byte)ImageOrientationPropertyEnum.Rotated180CW:
      magickImage.Rotate(0);
      break;
    case (byte)ImageOrientationPropertyEnum.Rotated90CW:
      //var temp = destWidth;
      //destWidth = destHeight;
      //destHeight = temp;
      magickImage.Rotate(270);
      //magickImage.AutoOrient();
      break;
  }

  if (destWidth > MaxImageWidth && destHeight != 0)
  {
    destWidth = MaxImageWidth;
    //var ratio = (float)magickImage.Width / magickImage.Height;
    var ratio = (float)magickImage.Width / magickImage.Height;
    destHeight = (int)(MaxImageWidth / ratio);
  }
  magickImage.FilterType = FilterType.Lanczos;
  magickImage.Resize(destWidth, destHeight); //if this is not commented out 

  // Rotate only if these rotation values are present.

  magickImage.AutoOrient();
  return magickImage;
}
```

**Reply by dlemstra** [reply]
You should only do the `AutoOrient` and remove the rest of your logic.

**Reply by shanji97** [reply]
Ok, I'll try as soon as I can get back to my work computer.

**Reply by shanji97** [reply]
Hi! I've removed everything, but still the problem persisted. I've decided to fix the issue so, that I manipulated the image directly (at System.Drawing.Bitmap level) at the upload and edited the exif tags accordingly after I rotated the image.
